src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.BoxedTypes.cs

```diff
@@ -112,13 +112,19 @@ public MethodDesc GetUnboxingThunk(MethodDesc targetMethod, ModuleDesc ownerModu
                 InstantiatedType boxedType = boxedTypeDefinition.MakeInstantiatedType(owningType.Instantiation);
                 MethodDesc thunk = GetMethodForInstantiatedType(thunkDefinition, boxedType);
                 //TODO: this might be triggered by a struct that implements an interface with a generic method
+                if (thunk.HasInstantiation)
```

Comment:
This could use a small cleanup. I would structure the code this way:

```csharp
MethodDesc thunk;
if (owningType != owningTypeDefinition)
{
  InstantiatedType boxedType = boxedTypeDefinition.MakeInstantiatedType(owningType.Instantiation);
  thunk = GetMethodForInstantiatedType(thunkDefinition, boxedType);
}
else
{
  thunk = thunkDefinition;
}

if (thunk.HasInstantiation)
  thunk = thunk.MakeInstantiatedMethod(targetMethod.Instantiation);

return thunk;
```

Avoids the duplicate `if (HasInstantiation)` logic.